package.json

```diff
@@ -14,8 +14,8 @@
     "node": ">=0.10.0"
   },
   "scripts": {
-    "pretest": "jshint .",
-    "test": "mocha -R spec --timeout 60000"
+    "test": "mocha -R spec --timeout 60000",
+    "posttest": "eslint --ignore-path .gitignore ."
```

Comment:
Its subjective and we made a choice for LoopBack projects. Doing it in posttest has the advantage of showing us test errors (rather than linting errors) in test when doing TDD. We then fix the linting errors *after* the main tests pass. I think Isaac from NPM does it the same way too (don't have link to article though, but you can see NPM using posttest in their examples too http://blog.npmjs.org/post/127671403050/testing-and-deploying-with-ordered-npm-run-scripts).

>Doing linting first is way better for CI speed - no need to wait 30 minutes just to find out that even though all the tests passed, you have a trailing whitespace somewhere.

Even on CI, I'd rather wait 30 mins to see the first logic test fail instead of getting back some linting error after 2 mins, then waiting 30 to see if my logic tests pass. Those can wait till I'm sure all the actual tests pass. Linting doesn't matter as much IMO.